cmd/query/app/server.go

```diff
@@ -193,37 +199,91 @@ type httpServer struct {
 
 var _ io.Closer = (*httpServer)(nil)
 
-func createHTTPServer(
-	ctx context.Context,
+func createHTTPRouter(
 	querySvc *querysvc.QueryService,
 	metricsQuerySvc querysvc.MetricsQueryService,
 	queryOpts *QueryOptions,
 	tm *tenancy.Manager,
 	telset telemetery.Setting,
-) (*httpServer, error) {
+) *mux.Router {
 	apiHandlerOptions := []HandlerOption{
 		HandlerOptions.Logger(telset.Logger),
 		HandlerOptions.Tracer(telset.TracerProvider),
 		HandlerOptions.MetricsQueryService(metricsQuerySvc),
 	}
-
 	apiHandler := NewAPIHandler(
 		querySvc,
 		tm,
 		apiHandlerOptions...)
+
 	r := NewRouter()
 	if queryOpts.BasePath != "/" {
 		r = r.PathPrefix(queryOpts.BasePath).Subrouter()
 	}
-
 	(&apiv3.HTTPGateway{
 		QueryService: querySvc,
 		TenancyMgr:   tm,
 		Logger:       telset.Logger,
 		Tracer:       telset.TracerProvider,
 	}).RegisterRoutes(r)
-
 	apiHandler.RegisterRoutes(r)
+
+	return r
+}
+
+func createHTTPServerOTEL(
+	ctx context.Context,
+	querySvc *querysvc.QueryService,
+	metricsQuerySvc querysvc.MetricsQueryService,
+	queryOpts *QueryOptions,
+	tm *tenancy.Manager,
+	telset telemetery.Setting,
+) (*httpServer, error) {
+	r := createHTTPRouter(querySvc, metricsQuerySvc, queryOpts, tm, telset)
```

Comment:
there is no requirement that `r` is actually a router, it just needs to be a handler. That means the following wrappers (token, recovery) can all be added in createHTTPRouter, as long as it returns a Handler.

Reply:
@yurishkuro `RegisterStaticHandler` later in the function requires the `mux.Router`

Reply:
static handler should be added in createHTTPRouter too, no reason to do it so late

Reply:
@yurishkuro the static handler closer exists on the custom `httpServer` type so we need to do it when initializing this type - its not part of the base handler.
```go
type httpServer struct {
	*http.Server
	staticHandlerCloser io.Closer
}
```

Reply:
we can return the closer from createHTTPRouter

Reply:
@yurishkuro i tried returning the closer but i was getting a weird error in the tests when i did that. returning both the handler and the router for now. 
```
2024/10/25 18:54:19 http: TLS handshake error from [::1]:53807: read tcp [::1]:53806->[::1]:53807: use of closed network connection
2024/10/25 18:54:19 http: TLS handshake error from [::1]:53810: remote error: tls: bad certificate
2024/10/25 18:54:19 http: TLS handshake error from [::1]:53813: read tcp [::1]:53812->[::1]:53813: use of closed network connection
2024/10/25 18:54:19 http: TLS handshake error from [::1]:53816: tls: client didn't provide a certificate
2024/10/25 18:54:19 http: TLS handshake error from [::1]:53817: tls: client didn't provide a certificate
2024/10/25 18:54:19 http: TLS handshake error from [::1]:53820: EOF
2024/10/25 18:54:19 http: TLS handshake error from [::1]:53821: client sent an HTTP request to an HTTPS server
2024/10/25 18:54:19 http: TLS handshake error from [::1]:53824: tls: failed to verify certificate: x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "Jaeger")
2024/10/25 18:54:19 http: TLS handshake error from [::1]:53825: tls: failed to verify certificate: x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "Jaeger")
2024/10/25 18:54:19 http: TLS handshake error from [::1]:53828: EOF
2024/10/25 18:54:20 http: TLS handshake error from [::1]:53829: client sent an HTTP request to an HTTPS server
PASS
goleak: Errors on successful test run: found unexpected goroutines:
[Goroutine 33 in state syscall, with syscall.syscall6 on top of the stack:
syscall.syscall6(0x0?, 0x0?, 0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
	/Users/mzaryab/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.23.2.darwin-arm64/src/runtime/sys_darwin.go:60 +0x50
golang.org/x/sys/unix.kevent(0x0?, 0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
	/Users/mzaryab/go/pkg/mod/golang.org/x/sys@v0.26.0/unix/zsyscall_darwin_arm64.go:275 +0x54
golang.org/x/sys/unix.Kevent(0x0?, {0x0?, 0x0?, 0x0?}, {0x140004cbe50?, 0x0?, 0x140004cbc98?}, 0x1051a1e10?)
	/Users/mzaryab/go/pkg/mod/golang.org/x/sys@v0.26.0/unix/syscall_bsd.go:397 +0x40
github.com/fsnotify/fsnotify.(*Watcher).read(0x0?, {0x140004cbe50, 0x0?, 0xa})
	/Users/mzaryab/go/pkg/mod/github.com/fsnotify/fsnotify@v1.7.0/backend_kqueue.go:777 +0x44
github.com/fsnotify/fsnotify.(*Watcher).readEvents(0x14000496070)
	/Users/mzaryab/go/pkg/mod/github.com/fsnotify/fsnotify@v1.7.0/backend_kqueue.go:547 +0x90
created by github.com/fsnotify/fsnotify.NewBufferedWatcher in goroutine 24
	/Users/mzaryab/go/pkg/mod/github.com/fsnotify/fsnotify@v1.7.0/backend_kqueue.go:184 +0x1fc
 Goroutine 34 in state select, with github.com/jaegertracing/jaeger/pkg/fswatcher.(*FSWatcher).watch on top of the stack:
github.com/jaegertracing/jaeger/pkg/fswatcher.(*FSWatcher).watch(0x140003ac080)
	/Users/mzaryab/Documents/opensource/jaeger/pkg/fswatcher/fswatcher.go:101 +0x84
created by github.com/jaegertracing/jaeger/pkg/fswatcher.New in goroutine 24
	/Users/mzaryab/Documents/opensource/jaeger/pkg/fswatcher/fswatcher.go:66 +0x100
]
```

Reply:
let's try to merge this first then rebase your PR on top of it (a good practice in any case - refactoring separately, actual changes separately) https://github.com/jaegertracing/jaeger/pull/6128